### PR TITLE
hypot tests for ax > sqrt(floatmax(ax)/2)

### PR DESCRIPTION
**Pull request by achuchmala**
Hi,
This is my first PR here :)
I've added tests for untested lines 664-666 in math.jl

### CODE REVIEW

File: test/math.jl [hunk 1]
```diff
@@ -1168,6 +1168,12 @@ end
             @test hypot(x, x*f, x*f) ≈ x * hypot(one(f), f, f) rtol=eps(T)
         end
     end
+    @testset "Real and Complex $T" for T in (Float16, Float32, Float64, BigFloat)
+        let x = sqrt(floatmax(T)/2 + 1)
```
Comment (simonbyrne) [inline test/math.jl:1172]: Thanks for the contribution!

Due to floating point rounding, `floatmax(T)/2 + 1 == floatmax(T)/2`, so this won't actually exercise the branch.

Should just test 
```suggestion
        let x = floatmax(T)
```

**Comment by achuchmala** [comment]
Thanks for review!

The accuracy and rounding with floating point numbers is indeed important thing to be aware of :)

I changed code according to your suggestion, however now I encounter issue with test. When running tests with make I got:

```
ach@ach-Vostro-7580:~/development/julia/packages/julia$ make test-math
    JULIA test/math
Test  (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
math       (1) |        started at 2020-11-05T01:59:33.535
math       (1) |         failed at 2020-11-05T02:00:11.506
Test Failed at /home/ach/development/julia/packages/julia/test/math.jl:1173
  Expression: #= /home/ach/development/julia/packages/julia/test/math.jl:1173 =# @inferred(hypot(x, complex(x / 4))) ≈ x * sqrt(17 / BigFloat(16))
   Evaluated: Inf16 ≈ 33759.98886255740458193770390071574268190526725735920411685601169650148618781252
```

this problem arises only for this test and Float16, when I run tests with make command, in REPL everything is OK.

I opened the issue regardinb this situation.

File: test/math.jl [hunk 2]
```diff
@@ -1168,6 +1168,12 @@ end
...
+            @test (@inferred hypot(x, complex(x/4))) ≈ x * sqrt(17/BigFloat(16))
```
Comment (simonbyrne) [inline test/math.jl:1173]: any reason you used `complex(x/4)` rather than just `x`?

Comment (achuchmala) [reply test/math.jl:1173]: its not any special reason, actually it could be just ```x```

File: test/math.jl [hunk 3]
```diff
@@ -1168,6 +1168,12 @@ end
             @test hypot(x, x*f, x*f) ≈ x * hypot(one(f), f, f) rtol=eps(T)
         end
     end
+    @testset "Real and Complex $T" for T in (Float16, Float32, Float64, BigFloat)
```
Comment (vchuravy) [inline test/math.jl:1171]: I would be okay with excluding `Float16` here and then having an explicit `@test_broken` case

**Comment by achuchmala** [comment]
I understand that now, when @simonbyrne fixed issue with Float16 this PR could be closed?
By the way - does adding tests that operate near the extremal values (e.g. as in this case for ```x = floatmax(T)/sqrt(2)```) to check how methods behave in these range of values makes sense?

**Comment by vtjnash** [comment]
Yeah, If I'm looking at the right place in https://coveralls.io/builds/38884059/source?filename=base%2Fmath.jl#L598, it looks like all lines are covered now. Thank you for your interest in helping us get to that point however!

**Pull request closed without merging**